index.js

```diff
@@ -81,12 +90,18 @@ class ServerlessPlugin {
    * @returns {Object}
    */
   parseEnvFiles(envFileNames) {
-    const envVarsArray = envFileNames.map((fileName) => {
-      const parsed = dotenv.config({ path: fileName })
-      return this.variableExpansion
-        ? dotenvExpand(parsed).parsed
-        : parsed.parsed
-    })
+    const dotenvParser = this.dotenvParserPath
+      ? require(this.dotenvParserPath)
+      : ({ path }) => {
```

Comment:
dotenvParser should get all the files at once. Otherwise, it can't do things like expand variables across multiple files at once.